lib/commands/battery.js

```diff
@@ -8,10 +10,26 @@ export default {
    * @this {import('../driver').XCUITestDriver}
    */
   async mobileGetBatteryInfo() {
-    return /** @type {BatteryInfo} */ (await this.proxyCommand('/wda/batteryInfo', 'GET'));
+    let batteryInfoFromShimService;
+    if (isIos18OrNewer(this.opts)) {
```

Comment:
I assume we should also add a check if the current device is real or if it is a simulator